lib/internal/modules/cjs/loader.js

```diff
@@ -303,19 +303,21 @@ function readPackage(requestPath) {
 
 function readPackageScope(checkPath) {
   const rootSeparatorIndex = checkPath.indexOf(path.sep);
-  let separatorIndex;
-  while (
-    (separatorIndex = checkPath.lastIndexOf(path.sep)) > rootSeparatorIndex
-  ) {
-    checkPath = checkPath.slice(0, separatorIndex);
+  let separatorIndex = checkPath.lastIndexOf(path.sep);
+  do {
     if (checkPath.endsWith(path.sep + 'node_modules'))
       return false;
+    if (rootSeparatorIndex === separatorIndex)
+      checkPath = checkPath.slice(0, rootSeparatorIndex + 1);
     const pjson = readPackage(checkPath);
     if (pjson) return {
+      data: pjson,
       path: checkPath,
-      data: pjson
     };
-  }
+    checkPath = checkPath.slice(0, separatorIndex);
```

Comment:
See if this works as mentioned above -

```suggestion
    checkPath = checkPath.slice(0, separatorIndex + 1);
```